### ISSUE TITLE
Refactor to use Redux hooks in payments-server components

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#9488]
React components in payment-server use Redux's magical `connect()` method to hook them up to the app-global store provided in React context.

Since we first wrote those components, Redux has released versions that come with [more modern React hooks](https://react-redux.js.org/next/api/hooks)

* i.e. [`useSelector()`](https://react-redux.js.org/next/api/hooks#useselector) and [`useDispatch()`](https://react-redux.js.org/next/api/hooks#usedispatch).

Might be a good exercise to refactor away the magical `connect()` to use the less-magical & more explicit selector & dispatch hooks.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-464)

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

@LZoog - not sure if you're looking for more "fun" refactoring opportunities, but this would be a good one 😀 (This is also not at all a priority)

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

@lmorchard I would definitely be down to do this. 😄 I'll leave it unassigned for now, though, since I've got a lot on my plate this quarter - when I have some time I'll circle back to see if it's still available.

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

Yeah, figured I'd bring it to your attention - but there is no urgency whatsoever

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

Did some tinkering over the weekend here, thinking that it might make doing #3241 easier.

Turns out, I was just yak shaving and it doesn't really help in particular. But, I pushed my work-in-progress to a branch on my repo ( https://github.com/lmorchard/fxa/compare/3020-payment-route-redux-hooks ) to pick up in the future - that, or someone else can look and steal ideas.
